app/controllers/tasks_controller.rb

```diff
@@ -1,4 +1,57 @@
 class TasksController < ApplicationController
+  before_action :get_list
+  before_action :get_task, only: [:show, :edit, :update, :destroy]
+
   def index
+    @tasks = @list.tasks
+  end
+
+  def new
+    @task = @list.tasks.build
+  end
+
+  def create
+    @task = @list.tasks.build(task_params)
+
+    if @task.save
+      redirect_to @task.list, notice: "Task `#{@task.title}` was successfully created."
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to @task.list, notice: "Task `#{@task.title}` was successfully updated."
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @task.destroy
+    redirect_to lists_path, notice: "Task `#{@task.title}` was successfully deleted."
+  end
+
+  private
+
+  def get_task
+    @task = Task.find_by(id: params[:id])
+
+    if @task.blank?
+      redirect_to @list, notice: "Task #{params[:id]} was not found."
+    end
+  end
+
+  def get_list
+    @list = List.find_by(id: params[:list_id])
```

Comment:
just a rails questions, how does it know about the `list_id` param? is it something generated by rails routes? 
cc @stejskalleos :D

Reply:
I think I can answer that,  @stejskalleos  correct me if I'm wrong. 
This feature is called [Nested Resources](https://guides.rubyonrails.org/routing.html#nested-resources), and it `allows you to capture this relationship in your routing`.
So the URL for any task will also include its list_id, for example: `/lists/{list_id}/tasks/{task_id}`